src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCreateCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -0,0 +1,283 @@
+        pdpAccountingEntry.setAmount(detailFileLine.getCashAdvanceAmount());
+        return pdpAccountingEntry;
+    }
+
+    
+    private PdpFeedPayeeIdEntry buildPdpFeedPayeeIdEntry(ConcurStandardAccountingExtractDetailLine detailFileLine) {
+        String defautPayeeIdType = getConcurBatchUtilityService().getConcurParameterValue(ConcurParameterConstants.DEFAULT_CONCUR_VALID_TRAVELER_STATUS_FOR_PDP_EMPLOYEE_CASH_ADVANCE_PROCESSING);
+        PdpFeedPayeeIdEntry payeeIdEntry = new PdpFeedPayeeIdEntry();
+        payeeIdEntry.setContent(detailFileLine.getEmployeeId());
+        if (getConcurBatchUtilityService().isValidTravelerStatusForProcessingAsPDPEmployeeType(defautPayeeIdType)) {
+            payeeIdEntry.setIdType(ConcurConstants.EMPLOYEE_PAYEE_STATUS_TYPE_CODE);
+        } else {
+            LOG.error("buildPdpFeedPayeeIdEntry: Invalid PayeeIdType detected from KFS System Parameter while building PDP output file:" + defautPayeeIdType);
+        }
+        return payeeIdEntry;
+    }
+
+    private PdpFeedHeaderEntry buildPdpFeedHeaderEntry(Date batchDate) {
+        PdpFeedHeaderEntry header = new PdpFeedHeaderEntry();
+        header.setChart(getConcurBatchUtilityService().getConcurParameterValue(ConcurParameterConstants.CONCUR_CUSTOMER_PROFILE_LOCATION));
+        header.setCreationDate(getConcurBatchUtilityService().formatDate_MMddyyyy(batchDate));
+        header.setSubUnit(getConcurBatchUtilityService().getConcurParameterValue(ConcurParameterConstants.CONCUR_CUSTOMER_PROFILE_SUB_UNIT));
+        header.setUnit(getConcurBatchUtilityService().getConcurParameterValue(ConcurParameterConstants.CONCUR_CUSTOMER_PROFILE_UNIT));
+        return header;
+    }
+
+    private PdpFeedTrailerEntry buildPdpFeedTrailerEntry(int totalDetailRecordCount, KualiDecimal totalDetailRecordAmount) {
+        PdpFeedTrailerEntry trailer = new PdpFeedTrailerEntry();
+        trailer.setDetailCount(totalDetailRecordCount);
+        trailer.setDetailTotAmt(totalDetailRecordAmount);
+        return trailer;
+    }
+
+    private boolean isDetailFileLineValidCashAdvanceRequest(ConcurStandardAccountingExtractDetailLine detailFileLine) {
+        return (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isValidCashAdvanceLine());
+    }
+
+    private boolean isDetailFileLineCashAdvanceUsedInExpenseReport(ConcurStandardAccountingExtractDetailLine detailFileLine) {
+        return (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isCashAdvanceUsedInExpenseReport());
+    }
+
+    private boolean isDetailFileLineDuplicateCashAdvanceRequest(ConcurStandardAccountingExtractDetailLine detailFileLine) {
+        return (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isDuplicatedCashAdvanceLine());
+    }
+
+    private boolean isDetailFileLineNotACashAdvanceRequest(ConcurStandardAccountingExtractDetailLine detailFileLine) {
```

Comment:
Chad: I will look at the method naming.

Regarding the empty collector file: The fix I have in my local changes that still need to be pushed to the feature branch is a check for a total number of lines to be output for the collector file.  If that total for lines to be output value is zero, then I do not create the collector file (data or done). I am seeing it work locally. Does that sound feasible to deal with the other issue you have seen?

Reply:
Yes, that sounds like it would work, as long as your solution still allows the PDP file to be created (if there are cases where an SAE file could result in several PDP rows but zero Collector rows). Thanks!

Reply:
@cah292 @jhulslander The latest changes for both the code and SQL are available for review.